cmd/lotus-seed/genesis.go

```diff
@@ -141,3 +149,153 @@ var genesisAddMinerCmd = &cli.Command{
+		}
+
+		if err := json.Unmarshal(b, &template); err != nil {
+			return xerrors.Errorf("unmarshal genesis template: %w", err)
+		}
+
+		entries, err := parseMultisigCsv(csvf)
+		if err != nil {
+			return xerrors.Errorf("parsing multisig csv file: %w", err)
+		}
+
+		for i, e := range entries {
+			if len(e.Addresses) != e.N {
+				return fmt.Errorf("entry %d had mismatch between 'N' and number of addresses", i)
+			}
+
+			msig := &genesis.MultisigMeta{
+				Signers:         e.Addresses,
+				Threshold:       e.M,
+				VestingDuration: monthsToBlocks(e.VestingMonths),
+				VestingStart:    0,
+			}
+
+			act := genesis.Actor{
+				Type:    genesis.TMultisig,
+				Balance: abi.TokenAmount(e.Amount),
+				Meta:    msig.ActorMeta(),
+			}
+
+			template.Accounts = append(template.Accounts, act)
+
+		}
+
+		b, err = json.MarshalIndent(&template, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		if err := ioutil.WriteFile(genf, b, 0644); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func monthsToBlocks(nmonths int) int {
+	days := uint64((365 * nmonths) / 12)
+	return int(days * 24 * 60 * 60 / build.BlockDelaySecs)
+}
+
+func parseMultisigCsv(csvf string) ([]GenAccountEntry, error) {
+	fileReader, err := os.Open(csvf)
+	if err != nil {
+		return nil, xerrors.Errorf("read multisig csv: %w", err)
+	}
+	r := csv.NewReader(fileReader)
+	records, err := r.ReadAll()
+	if err != nil {
+		return nil, xerrors.Errorf("read multisig csv: %w", err)
+	}
+	var entries []GenAccountEntry
+	for i, e := range records[1:] {
+		var addrs []address.Address
+		addrStrs := strings.Split(strings.TrimSpace(e[7]), ":")
+		for j, a := range addrStrs {
+			addr, err := address.NewFromString(a)
+			if err != nil {
+				return nil, xerrors.Errorf("failed to parse address %d in row %d (%q): %w", j, i, a, err)
+			}
+			addrs = append(addrs, addr)
+		}
+
+		balance, err := types.ParseFIL(e[2])
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse account balance: %w", err)
+		}
+
+		vesting, err := strconv.Atoi(e[3])
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse vesting duration for record %d: %w", i, err)
+		}
+
+		custodianID, err := strconv.Atoi(e[4])
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse custodianID in record %d: %w", i, err)
+		}
+		threshold, err := strconv.Atoi(e[5])
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse multisigM in record %d: %w", i, err)
+		}
+		num, err := strconv.Atoi(e[6])
+		if err != nil {
+			return nil, xerrors.Errorf("Number of addresses be integer: %w", err)
+		}
+		if e[0] != "1" {
+			return nil, xerrors.Errorf("record version must be 1")
+		}
+		entries = append(entries, GenAccountEntry{
+			Version:       1,
+			ID:            e[1],
+			Amount:        balance,
+			CustodianID:   custodianID,
+			VestingMonths: vesting,
+			M:             threshold,
+			N:             num,
+			Type:          e[8],
+			Sig1:          e[9],
+			Sig2:          e[10],
```

Comment:
Those seem to be unused, is this intentional?

Reply:
They will be used later, for now its just there to keep the input format consistent